VPI dirtiness tracking (#5065)
verilator/verilator: docs/guide/connecting.rst

For signal callbacks to work the main loop of the program must call
:code:`VerilatedVpi::callValueCbs()`.

Verilator also tracks when the model state has been modified via the VPI with

> a "dirty" flag.  This flag can be checked with :code:`VerilatedVpi::evalNeeded()`

Reviewer: ```suggestion
an :code:`evalNeeded` flag.  This flag can be checked with :code:`VerilatedVpi::evalNeeded()`
```

> and it can be cleared with :code:`VerilatedVpi::clearDirty()`.  Used together

Reviewer 2: Function names need updating.
